arwen/host/arwen.go

```diff
@@ -242,19 +244,23 @@ func (host *vmHost) RunSmartContractCall(input *vmcommon.ContractCallInput) (vmO
 		log.Error("RunSmartContractCall", "error", err)
 	}
 
+	ok := false
 	isUpgrade := input.Function == arwen.UpgradeFunctionName
 	if isUpgrade {
-		TryCatch(tryUpgrade, catch, "arwen.RunSmartContractUpgrade")
+		ok = TryCatch(tryUpgrade, catch, "arwen.RunSmartContractUpgrade")
 	} else {
-		TryCatch(tryCall, catch, "arwen.RunSmartContractCall")
+		ok = TryCatch(tryCall, catch, "arwen.RunSmartContractCall")
+	}
+
+	if ok && vmOutput != nil {
+		log.Trace("RunSmartContractCall end", "returnCode", vmOutput.ReturnCode, "returnMessage", vmOutput.ReturnMessage)
 	}
 
-	log.Trace("RunSmartContractCall end", "returnCode", vmOutput.ReturnCode, "returnMessage", vmOutput.ReturnMessage)
 	return
 }
 
 // TryCatch simulates a try/catch block using golang's recover() functionality
-func TryCatch(try TryFunction, catch CatchFunction, catchFallbackMessage string) {
+func TryCatch(try TryFunction, catch CatchFunction, catchFallbackMessage string) (ok bool) {
```

Comment:
I reverted the change, and kept the simpler version as it was before, but also kept the checks `if vmOutput != nil` (the ones that prevent a nil reference error within the following `log.Trace()` in case of `catch panic`). Is this ok?